src/vehicle.h

```diff
@@ -701,31 +704,54 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     // Loop through engines and generate noise and smoke for each one
     void noise_and_smoke( double load, double time = 6.0 );
 
-    // Calculate area covered by wheels and, optionally count number of wheels
-    float wheels_area (int *cnt = 0) const;
+    /**
+     * Calculates the sum of the area under the wheels of the vehicle.
+     * @param boat If true, calculates the area under "wheels" that allow swimming.
+     */
+    float wheel_area( bool boat ) const;
 
-    // Combined coefficient of aerodynamic and wheel friction resistance of vehicle, 0-1.0.
-    // 1.0 means it's ideal form and have no resistance at all. 0 -- it won't move
-    float k_dynamics () const;
+    /**
+     * Combined coefficient of aerodynamic and wheel friction resistance of vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the dynamics coefficient.
+     */
+    float k_dynamics() const;
 
-    // Components of the dynamic coefficient
+    /**
+     * Wheel friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the friction coefficient.
+     */
     float k_friction () const;
+
+    /**
+     * Air friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the aerodynamics coefficient.
+     */
     float k_aerodynamics () const;
 
-    // Coefficient of mass, 0-1.0.
-    // 1.0 means mass won't slow vehicle at all, 0 - it won't move
+    /**
+     * Mass coefficient of the vehicle.
+     * Depends on wheel area.
```

Comment:
> Documentation of dispersion and recoil fields doesn't describe how does dispersion and recoil work

How `dispersion` and `recoil` work is a frequent player question and should really have some documentation written to explain the system. An answer of *"it's an implementation detail"* would not be reasonable.

> Documenting how and where are the functions used is fragile and results in "documentation rot". Unless you mean something like "Affects speed and acceleration" (without specifics), because that's unlikely to change soon.

* Define `k_aerodynamics`, `k_friction` and `k_mass` as these are exposed directly to the user
* Document the valid range of returns for each `k_*` function and what the min/max means
* Provide an overall synopsis of how vehicles move and what affects that

The latter can be either part of the `vehicle` class documentation or added to the GH wiki.